src/Subscription.php

```diff
@@ -238,14 +238,20 @@ public function swapAndInvoice($plan)
      */
     public function swapNextCycle(string $plan)
     {
-        $new_plan = app(PlanRepository::class)::findOrFail($plan);
+        /** @var Plan $newPlan */
+        $newPlan = app(PlanRepository::class)::findOrFail($plan);
+
+        if ($this->cancelled()) {
```

Comment:
Can you move this inside the transaction?

Reply:
Of course, done.